Libraries/src/Amazon.Lambda.RuntimeSupport/Client/InternalClientAdapted.custom.cs

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+
+namespace Amazon.Lambda.RuntimeSupport
+{
+    internal partial interface IInternalRuntimeApiClient
+    {
+        /// <summary>
+        /// This is a copy of the generated Error2Async method but adds support for the unmodeled header `Lambda-Runtime-Function-XRay-Error-Cause`.
+        /// </summary>
+        /// <param name="awsRequestId"></param>
+        /// <param name="lambda_Runtime_Function_Error_Type"></param>
+        /// <param name="errorJson"></param>
+        /// <param name="xrayCause"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        System.Threading.Tasks.Task<SwaggerResponse<StatusResponse>> ErrorWithXRayCauseAsync(string awsRequestId, string lambda_Runtime_Function_Error_Type, string errorJson, string xrayCause, System.Threading.CancellationToken cancellationToken);
+    }
+
+    internal partial class InternalRuntimeApiClient : IInternalRuntimeApiClient
+    {
+        const int MAX_HEADER_SIZE_BYTES = 1024 * 1024;
+
+        /// <summary>
+        /// This is a copy of the generated Error2Async method but adds support for the unmodeled header `Lambda-Runtime-Function-XRay-Error-Cause`.
+        /// </summary>
+        /// <param name="awsRequestId"></param>
+        /// <param name="lambda_Runtime_Function_Error_Type"></param>
+        /// <param name="errorJson"></param>
+        /// <param name="xrayCause"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async System.Threading.Tasks.Task<SwaggerResponse<StatusResponse>> ErrorWithXRayCauseAsync(string awsRequestId, string lambda_Runtime_Function_Error_Type, string errorJson, string xrayCause, System.Threading.CancellationToken cancellationToken)
```

Comment:
Could we possibly reuse/extend relevant parts of `InternalRuntimeApiClient.ErrorAsync()`? (ref. https://github.com/aws/aws-lambda-dotnet/blob/master/Libraries/src/Amazon.Lambda.RuntimeSupport/Client/InternalClientAdapted.cs#L109)